### PR TITLE
Support the `.python-version` file

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: plugins/package-managers/python/src/main/kotlin/Pip.kt [hunk 1]
```diff
@@ -130,4 +130,10 @@ class Pip(
             }
         }.getOrThrow()
     }
+
+    private fun detectPythonVersion(workingDir: File): String? {
+        val pythonVersionFile = workingDir.resolve(".python-version")
+        if (!pythonVersionFile.isFile) return null
+        return pythonVersionFile.readText().trim().takeUnless { it.isEmpty() }
```
Comment (fviernau) [inline plugins/package-managers/python/src/main/kotlin/Pip.kt:137]: Do you have any reference for the format of the file? Would be nice to link it from a code comment.

Furthermore, would it make sense to make the check a tad more strict? e.g. return null if it's multiline string.

Comment (sschuberth) [reply plugins/package-managers/python/src/main/kotlin/Pip.kt:137]: > Do you have any reference for the format of the file?

No, see https://github.com/oss-review-toolkit/ort/issues/8366#issuecomment-1972676318.

> Furthermore, would it make sense to make the check a tad more strict? e.g. return null if it's multiline string.

Will do.

File: plugins/package-managers/python/src/main/kotlin/Pip.kt [hunk 2]
```diff
@@ -130,4 +130,12 @@ class Pip(
             }
         }.getOrThrow()
     }
+
+    private fun detectPythonVersion(workingDir: File): String? {
+        // While there seems to be no formal specification, a `.python-version` file seems to be supposed to simply
+        // conatin the plan version, see e.g. https://github.com/pyenv/pyenv/blob/21c2a3d/test/version.bats#L28.
```
Comment (fviernau) [inline plugins/package-managers/python/src/main/kotlin/Pip.kt:136]: typo: 
- splan -> plain
- conatin -> contain

maybe add a "just" before contain?

**Review by fviernau — CHANGES_REQUESTED** [review]

File: plugins/package-managers/python/src/main/kotlin/Pip.kt (hunk 2 above)
Comment (sschuberth) [reply plugins/package-managers/python/src/main/kotlin/Pip.kt]: > maybe add a "just" before contain?

As it should be either "just" or "simply", IMO, I've swapped out "simply" for "just".

**Review by fviernau — APPROVED** [review]

**Pull request merged**